tests/FakeItEasy.Specs/DummyCreationSpecs.cs

```diff
@@ -510,6 +510,22 @@ public void LazyCreationNonDummyable(Lazy<NonDummyable> dummy)
                 .x(() => dummy.Value.Should().BeNull());
         }
 
+        [Scenario]
+        public void LazyCreationIsLazy(Lazy<LazilyCreated> dummy)
+        {
+            "Given a type that tracks instance creation"
+                .x(() => LazilyCreated.IsInstanceCreated = false);

```

Comment:
I fear that there's a small chance that the `GenericDummyCreationSpecs` and `NonGenericDummyCreationSpecs` will run concurrently and `IsInstanceCreated` will be reset, spoiling the last assert.

Reply:
Ah, good catch. I thought there'd be no problem because this was only used in a single test, but I forgot it was actually two tests...
A simple fix would be to assign both `GenericDummyCreationSpecs` and `NonGenericDummyCreationSpecs` to the same xunit collection. Would that work for you, or do you have something more elaborate in mind?

Reply:
I added a `[Collection]` attribute to the base class (which applies it to the derived classes as well, since it's inherited), that should do it.

Reply:
Good one. I was considering atomic counting. Yours is superior.